plugins/modules/vcenter_ovf_libraryitem.py

```diff
@@ -261,93 +265,85 @@
-  register: lib_items
 
-- name: Create a new VM from the OVF
+# Due to an issue with the generated module code and the REST API limitations,
+# updating existing template descriptions must be done by removing the create_spec.name
+# attribute.
+#
+# Note that omitting the create_spec.name attribute will force the OVF template to be replaced.
+# Thus, the second task will always report as changed
+- name: Export the VM as an OVF on the library
   vmware.vmware_rest.vcenter_ovf_libraryitem:
-    ovf_library_item_id: '{{ (lib_items.value|selectattr("name", "equalto", "golden_image")|first).id }}'
-    state: deploy
+    session_timeout: 2900
+    source:
+        type: VirtualMachine
+        id: "{{ lookup('vmware.vmware_rest.vm_moid', 'My VM') }}"
     target:
-      resource_pool_id: "{{ lookup('vmware.vmware_rest.resource_pool_moid', '/my_dc/host/my_cluster/Resources') }}"
-    deployment_spec:
-      name: my_vm_from_ovf
-      accept_all_EULA: true
-      storage_provisioning: thin
+        library_id: 1111111-1111111-1111111-111111
+    create_spec:
+        name: Golden Image
+        description: Some unhelpful description
+        flags: []
+    state: present
+  register: _ovf_item
 
-- name: Create a new VM from the OVF and specify the host and folder
+- name: Update The Template Description
   vmware.vmware_rest.vcenter_ovf_libraryitem:
-    ovf_library_item_id: '{{ (lib_items.value|selectattr("name", "equalto", "golden_image")|first).id }}'
-    state: deploy
+    session_timeout: 2900
+    source:
+      type: VirtualMachine
+      id: "{{ lookup('vmware.vmware_rest.vm_moid', vm_name) }}"
     target:
-      resource_pool_id: "{{ lookup('vmware.vmware_rest.resource_pool_moid', '/my_dc/host/my_cluster/Resources') }}"
-      folder_id: "{{ lookup('vmware.vmware_rest.folder_moid', '/my_dc/vm') }}"
-      host_id: "{{ lookup('vmware.vmware_rest.host_moid', '/my_dc/host/my_cluster/esxi1.test/test_vm1') }}"
-    deployment_spec:
-      name: my_vm_from_ovf_on_a_host
-      accept_all_EULA: true
-      storage_provisioning: thin
+      library_id: "{{ library_id }}"
+      library_item_id: "{{ _ovf_item.id }}"
+    create_spec:
+      description: A much better description for this template
+      flags: []
 """
+
 RETURN = r"""
-# content generated by the update_return_section callback# task: Create a new VM from the OVF and specify the host and folder
+id:
+  description: The ID of the OVF template
+  returned: When the template is created
+  type: str
+  sample:
+    id: "3994f858-2d45-4dac-b407-0643a29308bd"
+
+# Sample shows values if OVF is created
 value:
-  description: Create a new VM from the OVF and specify the host and folder
+  description: Details about the OVF if it was created, or an ID if the OVF was updated
   returned: On success
-  sample:
-    error:
-      errors: []
-      information: []
-      warnings: []
-    resource_id:
-      id: vm-1078
-      type: VirtualMachine
-    succeeded: 1
   type: dict
+  sample:
```

Comment:
I got this result with `vmware.vmware.content_library_item_info` module, but `vmware.vmware_rest.vcenter_ovf_libraryitem` module returns:
```
    "_ovf_item": {
        "ansible_facts": {
            "discovered_interpreter_python": "/usr/bin/python3"
        },
        "changed": true,
        "failed": false,
        "value": {
            "error": {
                "errors": [],
                "information": [],
                "warnings": []
            },
            "ovf_library_item_id": "ea695665-307d-44da-b813-398969fc3903",
            "succeeded": true
        }
    }
```

Reply:
hm... Are you sure? Update task gave me the same result as Create task. Can you check it please one more time

Reply:
I updated the sample block. Hopefully its clearer that there are two possible return values